lib/services/services_linux.c

```diff
@@ -516,7 +516,7 @@ services__finalize_async_op(svc_action_t *op)
     if (op->interval_ms != 0) {
         // Recurring operations must be either cancelled or rescheduled
         if (op->cancel) {
-            op->status = PCMK_EXEC_CANCELLED;
+            services__set_result(op, op->rc, PCMK_EXEC_CANCELLED, NULL);
```

Comment:
Not having C++ would it make sense to think of things like PCMK_OCF_TRANSPARENT, PCMK_EXEC_TRANSPARENT as to be able to set single fields still keeping the number
of access-functions low. The compiler might even optimize that out with inline.

Reply:
I was wondering how best to deal with that. I think this particular case is fine since it's library-private API -- not even other pacemaker code can see this function, so low-level access is fine.

When we get to the point of having a public API for this sort of thing, it'll be more important to deal with. A new EXEC code is no big deal, but I'm wary of adding another OCF "extension". We treat it as an int, so maybe a negative number (to avoid any possibility of conflicting with future OCF additions) would be OK.

Reply:
Of course in the next chunk of work, I now have a use for services__set_result() in crm_resource, so it won't be library-private then ... but at least it's still internal API :)

Reply:
Yep was hesitant about PCMK_OCF_TRANSPARENT as we just deprecated some of the "extensions".
But I have no gold solution. A single value would make sense as it raises chances that it isn't hit somehow
accidentially. "extensions" could be negative and go to PCMK_OCF_INTERNAL_xxx - could still work with
services__set_result as long as it is internal API.

Reply:
Will think more about this and address it and the services__create_resource_action() refactoring in a later PR